firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift

```diff
@@ -334,10 +334,10 @@ extension BrowserViewController: ToolBarActionMenuDelegate {
                                                  textAlignment: .left)
             let toast = ButtonToast(viewModel: viewModel,
                                     theme: themeManager.currentTheme) { [weak self] isButtonTapped in
-                guard let strongSelf = self, let currentTab = strongSelf.tabManager.selectedTab else { return }
-                isButtonTapped ? strongSelf.addBookmark(
-                    url: currentTab.url?.absoluteString ?? "",
-                    title: currentTab.title
+                guard let self, let currentTab = tabManager.selectedTab else { return }
+                isButtonTapped ? self.addBookmark(
+                    url: bookmarkURL?.absoluteString ?? currentTab.url?.absoluteString ?? "",
+                    title: title ?? currentTab.title
```

Comment:
Q. should we be adding a url if it doesn't exist? 🤔

Reply:
Yes, we do. 

We have here two different cases in order to remove a bookmark.

1. From Hamburger Menu, where we should use the url from the current tab.
2. By using long tap on a link. For this case, we should pass the bookmark link and title, because are not the same as from the currentTab one.

This change was a request from @dataports too and in this case, is not a solution. 

Let me know if are necessary more details here. :)